membership-common/src/main/scala/com/gu/memsub/subsv2/reads/SubPlanReads.scala

```diff
@@ -132,20 +131,15 @@ object SubPlanReads {
           c: CatalogZuoraPlan,
       ): ValidationNel[String, SubscriptionPlan[P, ChargeList with SingleBenefit[B]]] = {
         (
-          ChargeListReads.readPaidCharge[B, BillingPeriod].read(c.benefits, z.charges.list.toList).map(\/.r[FreeCharge[B]].apply) orElse
-            ChargeListReads.readFreeCharge[B].read(c.benefits, z.charges.list.toList).map(\/.left)
-        ).flatMap(
-          _.fold(
-            free =>
-              freePlanReads(implicitly[SubPlanReads[P]], ChargeListReads.pure(free))
-                .read(p, z, c)
-                .map(identity[SubscriptionPlan[P, ChargeList with SingleBenefit[B]]]),
-            paid =>
-              paidPlanReads(implicitly[SubPlanReads[P]], ChargeListReads.pure(paid))
-                .read(p, z, c)
-                .map(identity[SubscriptionPlan[P, ChargeList with SingleBenefit[B]]]),
-          ),
-        )
```

Comment:
took me a minute to work this out - I think it might be using an Either in order to keep the correct types - so we don't need this.   I will revisit in the next PR when I remove Free/Paid distinction.